export async function promiseWithTimeout<T>(
  promise: Promise<T>,
  time: number,
  timeoutError: Error,
) {
  const timeoutPromise = new Promise((_, reject) => {
    setTimeout(() => reject(new Error()), time);
  });

  return Promise.race([promise, timeoutPromise])
    .then((result) => result as T)
    .catch(() => {
      throw timeoutError;
    });
}
